Libs/MRML/Widgets/qMRMLVolumeThresholdWidget.cxx

```diff
@@ -263,6 +264,10 @@ void qMRMLVolumeThresholdWidget::updateWidgetFromMRMLDisplayNode()
   // even if the values have been just set programmatically/automatically.
   bool wasBlocking = d->VolumeThresholdRangeWidget->blockSignals(true);
 
+  double range[2];
+  d->scalarRange(d->VolumeDisplayNode, range);
```

Comment:
Scalar range is cached in vtkImageData but the first time it is computed it is a relatively expensive operation (min/max operation on all voxel values). It can significantly slow down live volume display and volume sequence replay. Make sure that this method is only called if the Volumes module is active and Display section is open.

It is similar to how `qSlicerScalarVolumeDisplayWidget::updateHistogram()` is protected from being called all the time: it is only updated continuously when the histogram section is shown and not collapsed.

Reply:
I've updated the logic so that `scalarRange` is only called once as part of `updateRangeForVolumeDisplayNode` of the qMRMLVolumeWidget class. This is how it is currently in the Slicer `master` branch.